src/aux-server/aux-web/aux-auth/site/AuthApp/AuthApp.ts

```diff
@@ -127,47 +147,103 @@ export default class AuthApp extends Vue {
+        this.showEnterRecordNameError = false;
+        this.showInvalidRecordNameError = false;
     }
 
     async createStudio() {
-        this.showCreateStudio = false;
-        const comId = authManager.getComIdFromUrl();
-        const result = await authManager.client.createStudio({
-            displayName: this.studioName,
-            ownerStudioComId: comId,
-        });
-        await this.loadStudios();
+        try {
+            this.showEnterStudioNameError = false;
+            this.showCreateStudioError = false;
+            this.errorMessage = '';
+
+            if (!this.studioName) {
+                this.showEnterStudioNameError = true;
+                return;
+            }
+
+            const comId = authManager.getComIdFromUrl();
+            const response = await authManager.client.createStudio({
+                displayName: this.studioName,
+                ownerStudioComId: comId,
+            });
+
+            if (response.success == false) {
+                this.errorMessage = response.errorMessage;
+                if (response.errorCode === 'invalid_key') {
+                    this.showCreateStudioError = true;
+                }
+                return;
+            }
+
+            this.showCreateStudio = false;
+            await this.loadStudios();
+        } catch (error) {
+            console.error('Studio creation failed:', error);
+            this.errorMessage = error.message;
+            this.showCreateStudioError = true;
+        }
     }
 
     async createRecord() {
-        this.showCreateRecord = false;
-        let request: Omit<CreateRecordRequest, 'userId'> = {
-            recordName: this.recordName,
-        };
-
-        if (this.createRecordStudioId) {
-            request['studioId'] = this.createRecordStudioId;
-        } else {
-            request['ownerId'] = authManager.userId;
-        }
+        try {
+            this.showEnterRecordNameError = false;
+            this.showInvalidRecordNameError = false;
+            this.errorMessage = '';
+
+            if (!this.recordName) {
+                this.showEnterRecordNameError = true;
+                return;
+            }
 
-        await authManager.client.createRecord(request);
+            let request: Omit<CreateRecordRequest, 'userId'> = {
+                recordName: this.recordName,
+                ...(this.createRecordStudioId
+                    ? { studioId: this.createRecordStudioId }
+                    : { ownerId: authManager.userId }),
+            };
 
-        if (this.createRecordStudioId) {
-            const studio = this.studios.find(
-                (s) => s.studioId === this.createRecordStudioId
-            );
-            if (studio) {
-                await this.onExpandStudio(studio);
+            if (this.createRecordStudioId) {
+                request['studioId'] = this.createRecordStudioId;
+            } else {
+                request['ownerId'] = authManager.userId;
+            }
```

Comment:
@TroyceGowdy This code seems to do the same thing as the object initializer above it